addons/website_sale/migrations/10.0.1.0/post-migration.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+@openupgrade.migrate()
+def migrate(env, version):
+    cr = env.cr
+    sql = """select pricelist_id, website_id, selectable from 
+          website_pricelist_openupgrade_10"""
+    cr.execute(sql)
+    for pricelist_id , website_id, selectable in  cr.fetchall():
+        env['product.pricelist'].search([('id', '=', pricelist_id)]).write({
+            'website_id': website_id,
+            'selectable': selectable
+        })
+    # now that all pricelists have a their website the field pricelist_ids in
```

Comment:
website_id has a default  self.env['website'].search([], limit=1)  ,but is not a required field.
So it should be empty sometimes?
I had started to code your proposal by calling _default_website(), but technically if the original db has no website ID so it should remain.
You can make null.

do you agree?